tests/TestCase/Validation/ValidatorTest.php

```diff
@@ -1430,6 +1448,38 @@ public function testNotEmptyAsArray(): void
         $this->assertEquals($expected, $errors);
     }
 
+    /**
+     * Tests the notEmpty method with an array containing an integer key
+     */
+    public function testNotEmptyAsArrayWithIntegerKey(): void
+    {
+        $validator = new Validator();
+
+        $this->deprecated(function () use ($validator): void {
+            $validator->notEmpty([
+                0,
+                1 => [
+                    'when' => false,
+                ],
+                'another_field',
+            ], 'Not empty', true);
+        });
+
+        $this->assertTrue($validator->field((string)0)->isEmptyAllowed());
```

Comment:
```suggestion
        $this->assertTrue($validator->field('0')->isEmptyAllowed());
```
Any reason to not use string literals?

Reply:
Mostly just to display this is a special case and not the way it is (normally) supose to be.
I'm fine to update the calls.

Reply:
It seems like a long way to get to a string, and future us will likely have less context on why this is special.